shark/src/main/java/shark/HeapAnalyzer.kt

```diff
@@ -86,6 +87,20 @@ class HeapAnalyzer constructor(
       )
     }
 
+    if (obfuscationChecker.isCodeObfuscated() && proguardMapping == null) {
```

Comment:
This doesn't always work. The obfuscation checker is based on the runtime, and the code that runs the analysis might not be the same as the code that was used for the hprof. This should check inside the hprof.